TradeItIosTicketSDK2/TradeItTransactionsViewController.swift

```diff
@@ -0,0 +1,97 @@
+import UIKit
+import MBProgressHUD
+import PromiseKit
+
+class TradeItTransactionsViewController: TradeItViewController, TradeItTransactionsTableDelegate {
+    
+    let alertManager = TradeItAlertManager()
+    var transactionsTableViewManager: TradeItTransactionsTableViewManager?
+    
+    @IBOutlet weak var transactionsTable: UITableView!
+    
+    var linkedBrokerAccount: TradeItLinkedBrokerAccount?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        guard let linkedBrokerAccount = self.linkedBrokerAccount else {
+            preconditionFailure("TradeItIosTicketSDK ERROR: TradeItTransactionsViewController loaded without setting linkedBrokerAccount.")
+        }
+        self.transactionsTableViewManager = TradeItTransactionsTableViewManager(linkedBrokerAccount: linkedBrokerAccount)
+        self.transactionsTableViewManager?.delegate = self
+        self.transactionsTableViewManager?.transactionsTable = transactionsTable
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        self.loadTransactions()
+    }
+    
+    // MARK: IBAction
+    
+    @IBAction func filterButtonWasTapped(_ sender: Any) {
+        //TODO: https://www.pivotaltracker.com/story/show/148168413
+    }
+    // MARK: TradeItTransactionsTableDelegate
+    
+    func refreshRequested(onRefreshComplete: @escaping () -> Void) {
+        guard let linkedBrokerAccount = self.linkedBrokerAccount
+            , let linkedBroker = linkedBrokerAccount.linkedBroker else {
+                preconditionFailure("TradeItIosTicketSDK ERROR: TradeItTransactionsViewController loaded without setting linkedBrokerAccount.")
+        }
+        func authenticatePromise() -> Promise<Void>{
+            return Promise<Void> { fulfill, reject in
+                linkedBroker.authenticateIfNeeded(
+                    onSuccess: fulfill,
+                    onSecurityQuestion: { securityQuestion, answerSecurityQuestion, cancelSecurityQuestion in
+                        self.alertManager.promptUserToAnswerSecurityQuestion(
+                            securityQuestion,
+                            onViewController: self,
+                            onAnswerSecurityQuestion: answerSecurityQuestion,
+                            onCancelSecurityQuestion: cancelSecurityQuestion
+                        )
+                },
+                    onFailure: reject
+                )
+            }
+        }
```

Comment:
I think we have this in a bunch of places now. Maybe we can extract it to an internal method on the `linkedBroker`?

Reply:
Because of the security question we have a reference to the alertManager and the view:
```
alertManager.promptUserToAnswerSecurityQuestion(
                        securityQuestion,
                        onViewController: self,
                        onAnswerSecurityQuestion: answerSecurityQuestion,
                        onCancelSecurityQuestion: cancelSecurityQuestion
                    )
```
How we could refactor this in the linkedBroker ?